### PR TITLE
Added more error routes for comprehensive error handling

### PR DESCRIPTION
**Pull request by TaylorSmith28**

### CODE REVIEW

**Comment by TaylorSmith28** [comment]
Do not merge this yet unless someone is planning on making changes to the routes/error.py file.

**Comment by TaylorSmith28** [comment]
This branch is ready to be merged, updated machines/routes for put implementation, more error route handling, and added more tests.

**Review by haydentinker — APPROVED** [review]
Thank you for updating error handlers/things in machine.py and making our tests more comprehensive. Looks good overall.

**Pull request merged**
